### PR TITLE
Do not crash on invalid protocol messages

### LINKED ISSUE #2296: Not crash on malformed messages and prepare for message versioning
## Problem Definition

Related to #2161 

While upgrading to [v0.5.0](https://github.com/raiden-network/raiden/releases/tag/v0.5.0) there was a minor change done to the Raiden messages. The `channel` attribute of the message was renamed to `channel_identifier`. All tests were green but when playing on the testnet with peers who were still before the big merge of the channel identifier changes @kelsos reported a crash of his node.

```
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run
  File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/matrix_client/room.py", line 308, in _put_event
    listener['callback'](self, event)
  File "/home/kelsos/development/github/raiden-network/1/raiden/network/transport/matrix.py", line 485, in _handle_message
    message = message_from_dict(message_dict)
  File "/home/kelsos/development/github/raiden-network/1/raiden/messages.py", line 139, in from_dict
    return klass.from_dict(data)
  File "/home/kelsos/development/github/raiden-network/1/raiden/messages.py", line 1309, in from_dict
    channel_identifier=data['channel_identifier'],
KeyError: 'channel_identifier'
2018-08-10T13:50:13Z <RaidenGreenlet "Greenlet-0" at 0x7fc5907b5748: <bound method Room._put_event of <Room id="!ZbsJLMCIDoWiMbMUdg:transport03.raiden.network" alias="#raiden_ropsten_0x02f4b6bc65561a792836212ebc54434db0ab759a_0x82641569b2062b545431cf6d7f0a418582865ba7:transport03.raiden.network">>({'origin_server_ts': 1533909010478, 'sender': '@0x)> failed with KeyError

FATAL: An unexpected exception occured. A traceback has been written to /tmp/raiden-exception-2018-08-10T13-50tqc8lerx.txt
'channel_identifier'

➜ cat /tmp/raiden-exception-2018-08-10T13-50tqc8lerx.txt
Traceback (most recent call last):
  File "/home/kelsos/development/github/raiden-network/1/raiden/ui/cli.py", line 992, in _run_app
    event.wait()
  File "src/gevent/event.py", line 240, in gevent._event.Event.wait
  File "src/gevent/event.py", line 140, in gevent._event._AbstractLinkable._wait
  File "src/gevent/event.py", line 117, in gevent._event._AbstractLinkable._wait_core
  File "src/gevent/event.py", line 119, in gevent._event._AbstractLinkable._wait_core
  File "src/gevent/_greenlet_primitives.py", line 59, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/_greenlet_primitives.py", line 59, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/_greenlet_primitives.py", line 63, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/__greenlet_primitives.pxd", line 35, in gevent.__greenlet_primitives._greenlet_switch
KeyError: 'channel_identifier'
```

This was essentially on the `from_dict()` [function](https://github.com/raiden-network/raiden/blob/b29f08e57dcd6961d74d12578e8b2c2864930db9/raiden/messages.py?utf8=%E2%9C%93#L1309) of the `LockedTransfer` message.

Indeed when looking at the message with pdb:

```
 > /home/kelsos/development/github/raiden-network/1/raiden/messages.py(1324)from_dict()
 -> message.signature = decode_hex(data['signature'])
 (Pdb) print data
 *** SyntaxError: Missing parentheses in call to 'print'. Did you mean print(data)?
 (Pdb) print(data)
 {'type': 'LockedTransfer', 'chain_id': 3, 'message_identifier': 18271163750950353082, 'payment_identifier': 1533742814198, 'nonce': 5, 'token_network_address': '0x7692624c0c43285d51c0dddfb2a6388f5002762e', 'token': '0x0f114a1e9db192502e7856309cc899952b3db1ed', 'channel': '0x2049836f9260624a99b06c6bf06ddc5ca1dfc601216129246f475f4256154813', 'transferred_amount': 10, 'locked_amount': 5, 'recipient': '0x82641569b2062b545431cf6d7f0a418582865ba7', 'locksroot': '0x37889b21a5989b23ae5a9429710ec8caadb664d953cd117c1de6f1a52f2fe27a', 'lock': {'type': 'Lock', 'amount': 5, 'expiration': 3797904, 'secrethash': '0xfd5834561a1a309b0e1cf24fed422989da5dcec83a54aa30e960d637c9f1c81a'}, 'target': '0x82641569b2062b545431cf6d7f0a418582865ba7', 'initiator': '0x02f4b6bc65561a792836212ebc54434db0ab759a', 'fee': 0, 'signature': '0xb284376b0323cf39fa1e7d1d6ab09b7b3f069029d85bbcd2624f3c86bc2e3dc24e12cd1020f90e6f302e3e88d47cfb39da710b6675617760181329e7270c73d81c'}
 (Pdb) c
 Traceback (most recent call last):
   File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run
   File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/matrix_client/room.py", line 308, in _put_event
     listener['callback'](self, event)
   File "/home/kelsos/development/github/raiden-network/1/raiden/network/transport/matrix.py", line 485, in _handle_message
     message = message_from_dict(message_dict)
   File "/home/kelsos/development/github/raiden-network/1/raiden/messages.py", line 139, in from_dict
     return klass.from_dict(data)
   File "/home/kelsos/development/github/raiden-network/1/raiden/messages.py", line 1324, in from_dict
     """ A special LockedTransfer sent from a payee to a payer indicating that
 UnboundLocalError: local variable 'message' referenced before assignment
 2018-08-10T14:56:17Z <RaidenGreenlet "Greenlet-0" at 0x7fecf8516548: <bound method Room._put_event of <Room id="!ZbsJLMCIDoWiMbMUdg:transport03.raiden.network" alias="#raiden_ropsten_0x02f4b6bc65561a792836212ebc54434db0ab759a_0x82641569b2062b545431cf6d7f0a418582865ba7:transport03.raiden.network">>({'origin_server_ts': 1533912527415, 'sender': '@0x)> failed with UnboundLocalError
```

you can easily notice that the message he received is in the old format with `channel` instead of `channel_identifier.

## Goal of the issue

Prepare for message versioning as outlined in #2161 and not crash on old/invalid messages.

## Task

- Since we may add fields to messages in the future and in order to be forward compatible we need to remove [this](https://github.com/raiden-network/raiden/blob/master/raiden/encoding/format.py#L106) check from the UDP transport

- Make sure that we don't crash on malformed or old messages.

### PR DESCRIPTION
**Pull request by LefterisJP**
Work in progresss

Fix #2296

### CODE REVIEW

**Comment by LefterisJP** [comment]
note to self: Don't forget to add changelog entry

File: raiden/network/transport/matrix.py [hunk 1]
```diff
@@ -514,7 +537,15 @@ def _handle_message(self, room, event):
                     peer_address=pex(peer_address),
                     exception=ex,
                 )
-                return
+                return False
+            except InvalidProtocolMessage as ex:
+                self.log.warning(
```
Comment (rakanalh) [inline raiden/network/transport/matrix.py]: I think this should be a call to `error` ... unless we just want to warn and move on, then the UDP transport should also do this.

File: raiden/network/transport/udp/udp_transport.py [hunk 2]
```diff
@@ -444,9 +445,18 @@ def receive(self, messagedata: bytes):
...
                 length=len(messagedata),
             )
-            return
+            return False
 
-        message = decode(messagedata)
+        try:
+            message = decode(messagedata)
+        except InvalidProtocolMessage as e:
+            log.error(
```
Comment (rakanalh) [inline raiden/network/transport/udp/udp_transport.py]: made a comment about `error` vs `warning` above.

File: raiden/network/transport/udp/udp_transport.py [hunk 3]
```diff
@@ -444,9 +445,18 @@ def receive(self, messagedata: bytes):
...
+                'INVALID PROTOCOL MESSAGE',
```
Comment (rakanalh) [inline raiden/network/transport/udp/udp_transport.py]: ALL CAPS?

File: raiden/network/transport/matrix.py (hunk 1 above)
Comment (LefterisJP) [reply raiden/network/transport/matrix.py]: yep we should just warn

**Pull request merged**